### ISSUE TITLE
Name collision between top-level property and definition

### ISSUE THREAD

**Issue by michaelmior** [corvus-dotnet/Corvus.JsonSchema#464]
There seems to be a name collision between the property `builder` and the definition `builder`.

```json
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "properties": {
    "builder": {}
  },
  "definitions": {
    "builder": {}
  }
}
```

**Comment by mwadams** [comment]
Note to self: This is the same disambiguation issue as #465, in a different context.

**Comment by michaelmior** [comment]
@mwadams Feel free to close anything that's duplicated if it's helpful for organizing things. It's not always obvious to me what might be caused by the same underlying issue.

**Comment by mwadams** [comment]
This one _is_ still present in v4.0.0-preview.24!

**Comment by mwadams** [comment]
In v4.0.0-preview.24 the disambiguation is being handled correctly, but we have a variable name `builder` in the implementation of the `Create()` method that is not being protected against a naming collision in the code generator.

**Comment by mwadams** [comment]
Generate unique variable name to avoid collision with property. This will be available in 4.0.0-preview.25 later today.

**Comment by michaelmior** [comment]
I get different compilation errors now.

```
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(659,54): error CS0117: 'JsonValueHelpers' does not contain a definition for 'GetHashCodeForNumber' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(660,54): error CS0117: 'JsonValueHelpers' does not contain a definition for 'GetHashCodeForString' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(663,52): error CS0117: 'JsonValueHelpers' does not contain a definition for 'NullHashCode' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(664,35): error CS0117: 'JsonValueHelpers' does not contain a definition for 'UndefinedHashCode' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Object.cs(91,48): error CS1061: 'JsonElement' does not contain a definition for 'GetPropertyCount' and no accessible extension method 'GetPropertyCount' accepting a first argument of type 'JsonElement' could be found (are you missing a using directive or an assembly reference?) [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Validate.cs(80,51): error CS1501: No overload for method 'WithResult' takes 3 arguments [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Validate.cs(91,30): error CS1501: No overload for method 'NameEquals' takes 2 arguments [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Validate.cs(96,41): error CS1061: 'ValidationContext' does not contain a definition for 'PushValidationLocationReducedPathModifierAndProperty' and no accessible extension method 'PushValidationLocationReducedPathModifierAndProperty' accepting a first argument of type 'ValidationContext' could be found (are you missing a using directive or an assembly reference?) [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]

Build FAILED.

/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(659,54): error CS0117: 'JsonValueHelpers' does not contain a definition for 'GetHashCodeForNumber' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(660,54): error CS0117: 'JsonValueHelpers' does not contain a definition for 'GetHashCodeForString' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(663,52): error CS0117: 'JsonValueHelpers' does not contain a definition for 'NullHashCode' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.cs(664,35): error CS0117: 'JsonValueHelpers' does not contain a definition for 'UndefinedHashCode' [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Object.cs(91,48): error CS1061: 'JsonElement' does not contain a definition for 'GetPropertyCount' and no accessible extension method 'GetPropertyCount' accepting a first argument of type 'JsonElement' could be found (are you missing a using directive or an assembly reference?) [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Validate.cs(80,51): error CS1501: No overload for method 'WithResult' takes 3 arguments [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Validate.cs(91,30): error CS1501: No overload for method 'NameEquals' takes 2 arguments [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
/home/mmior/apps/jsonschema-benchmark/implementations/corvus/Schema.Validate.cs(96,41): error CS1061: 'ValidationContext' does not contain a definition for 'PushValidationLocationReducedPathModifierAndProperty' and no accessible extension method 'PushValidationLocationReducedPathModifierAndProperty' accepting a first argument of type 'ValidationContext' could be found (are you missing a using directive or an assembly reference?) [/home/mmior/apps/jsonschema-benchmark/implementations/corvus/bench.csproj]
    0 Warning(s)
    8 Error(s)

Time Elapsed 00:00:00.95
```
